libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -329,6 +352,13 @@ constructor(
         ThreadController.cancelAllUICoroutines()
     }
 
+    /**
+     * Replace default [MapboxRerouteController]
+     */
+    fun setRerouteController(rerouteController: RerouteController) {
```

Comment:
Could you also add a `getRerouteController` so that a user could subscribe to state events like mentioned above?

Reply:
by this way we give to developer opportunity to invoke default `reroute` and `interrupt reroute`. Let's concentrate to limit and check how it works for us, may be we'd change it a little soon

Reply:
I think that's totally fine. They could've done that when passing their own instance anyway, and also giving developers an option to cancel the request is another benefit.

Reply:
with https://github.com/mapbox/mapbox-navigation-android/pull/3184#discussion_r446108034 it's a good point 👍